executor/adapter.go

```diff
@@ -424,7 +424,6 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 		oriStats, _ := sctx.GetSessionVars().GetSystemVar(variable.TiDBBuildStatsConcurrency)
 		oriScan := sctx.GetSessionVars().DistSQLScanConcurrency()
 		oriIndex := sctx.GetSessionVars().IndexSerialScanConcurrency()
-		oriIso, _ := sctx.GetSessionVars().GetSystemVar(variable.TxnIsolation)
 		terror.Log(sctx.GetSessionVars().SetSystemVar(variable.TiDBBuildStatsConcurrency, "1"))
 		sctx.GetSessionVars().SetDistSQLScanConcurrency(1)
 		sctx.GetSessionVars().SetIndexSerialScanConcurrency(1)
```

Comment:
Since it is set below at line 431/430 like this:
`terror.Log(sctx.GetSessionVars().SetSystemVar(variable.TxnIsolation, ast.ReadCommitted))`
Should it not be restored upon return?
Or should InRestrictedSSQL always run in ReadCommitted?

Reply:
Sorry, I missed it. Actually the previous `analyze` should have used https://github.com/pingcap/tidb/blob/619c4720059ea619081b01644ef3084b426d282f/kv/kv.go#L449, which is not the same as isolation level = 'READ-COMMITTED'. So, I think that the isolation setting here should be meaningless. However, I will confirm this further.